### ISSUE TITLE
[General Issue]PP-LiteSeg训练OhemcrossentropyLoss报错

### ISSUE THREAD

**Issue by JiehangXie** [PaddlePaddle/PaddleSeg#2542]
1. PaddleSeg version: 2.6
 2. PaddlePaddle version: 2.3.2
 3. Operation system: Linux
 4. Python version: 3.7
 5. CUDA/cuDNN version: 11.2

报错如下，经检查num_classes没问题，应该往哪个方向排查问题？
![7FI9_HZBHOOX(LDHR`C@JNC](https://user-images.githubusercontent.com/51190264/190364503-6d777b1f-c845-4268-befb-b5e19a22391d.png)

**Comment by JiehangXie** [comment]
batch_size: 2
iters: 160000

model:
  type: PPLiteSeg
  backbone:
    type: STDC2
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz

train_dataset:
  type: Dataset
  dataset_root: data/aeroscapes
  train_path: data/train.txt
  num_classes: 11
  transforms:
    - type: ResizeStepScaling
      min_scale_factor: 0.5
      max_scale_factor: 2.0
      scale_step_size: 0.25
    - type: RandomPaddingCrop
      crop_size: [1024, 512]
    - type: RandomHorizontalFlip
    - type: RandomDistort
      brightness_range: 0.5
      contrast_range: 0.5
      saturation_range: 0.5
    - type: Normalize
  mode: train

val_dataset:
  type: Dataset
  dataset_root: data/aeroscapes
  val_path: data/val.txt
  num_classes: 11
  transforms:
    - type: Normalize
  mode: val

test_config:
  aug_eval: True
  scales: 1.0

optimizer:
  type: sgd
  momentum: 0.9
  weight_decay: 5.0e-4

lr_scheduler:
  type: PolynomialDecay
  warmup_iters: 1000
  warmup_start_lr: 1.0e-5
  end_lr: 0.005
  learning_rate: 0.005

loss:
  types:
    - type: OhemCrossEntropyLoss
      min_kept: 65536   # batch_size * 1024 * 512 // 16
    - type: OhemCrossEntropyLoss
      min_kept: 65536
    - type: OhemCrossEntropyLoss
      min_kept: 65536
  coef: [1, 1, 1]

**Comment by Bobholamovic** [comment]
有可能是OOM错误导致的。建议可以使用如下指令观察一下程序运行时的显存占用情况：

```bash
watch -n 1 nvidia-smi
```

**Comment by JiehangXie** [comment]
> 有可能是OOM错误导致的。建议可以使用如下指令观察一下程序运行时的显存占用情况：
> 
> ```shell
> watch -n 1 nvidia-smi
> ```

显存没有溢出，bs为2，也在A100环境下试过，依然报错

**Comment by Bobholamovic** [comment]
这样的话请检查标签中是否存在无效的索引值。例如，在[此位置](https://github.com/PaddlePaddle/PaddleSeg/blob/d2f566782ac03b2acae0051df92718af7f68b2fe/paddleseg/models/losses/ohem_cross_entropy_loss.py#L57)打印标签信息`print(label.max(), label.min(), label.shape, label.dtype)`以实施检查。

**Comment by JiehangXie** [comment]
> 这样的话请检查标签中是否存在无效的索引值。例如，在[此位置](https://github.com/PaddlePaddle/PaddleSeg/blob/d2f566782ac03b2acae0051df92718af7f68b2fe/paddleseg/models/losses/ohem_cross_entropy_loss.py#L57)打印标签信息`print(label.max(), label.min(), label.shape, label.dtype)`以实施检查。

谢谢，检查出来了，新版本没有默认ignore 255吗？

**Comment by Bobholamovic** [comment]
ignore 255应该是有做的，在如下位置：
https://github.com/PaddlePaddle/PaddleSeg/blob/c608005d4de4a10ae9ec9be76c53951cf658f758/paddleseg/models/losses/ohem_cross_entropy_loss.py#L59

如果标签中包含无效值的话，建议可以查看一下除255以外的label中的其它索引值是否存在异常，使用如下指令：
`valid_label=label[label!=255]; print(valid_label.max(), valid_label.min())`

**Comment by JiehangXie** [comment]
> ignore 255应该是有做的，在如下位置：
> 
> https://github.com/PaddlePaddle/PaddleSeg/blob/c608005d4de4a10ae9ec9be76c53951cf658f758/paddleseg/models/losses/ohem_cross_entropy_loss.py#L59
> 
> 如果标签中包含无效值的话，建议可以查看一下除255以外的label中的其它索引值是否存在异常，使用如下指令： `valid_label=label[label!=255]; print(valid_label.max(), valid_label.min())`

OK，这里检查出来了，有一个最大值11，所以是12个label，谢谢了！！

**Comment by djl00** [comment]
Hello, I would like to ask how testing is conducted on the cityscapes dataset?
